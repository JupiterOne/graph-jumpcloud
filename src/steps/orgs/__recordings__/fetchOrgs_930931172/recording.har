{
  "log": {
    "_recordingName": "fetchOrgs",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "68692ed96e085eab550b21a15bbb26da",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-org-id",
              "value": ""
            },
            {
              "_fromType": "array",
              "name": "x-api-key",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "contenttype",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "console.jumpcloud.com"
            }
          ],
          "headersSize": 320,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://console.jumpcloud.com/api/organizations"
        },
        "response": {
          "bodySize": 288,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 288,
            "text": "{\"totalCount\":1,\"results\":[{\"_id\":\"60259a2ad81e1a474d4e7c83\",\"id\":\"60259a2ad81e1a474d4e7c83\",\"displayName\":null,\"logoUrl\":null,\"created\":\"2021-02-11T20:57:14.837Z\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "nginx"
            },
            {
              "name": "date",
              "value": "Fri, 12 Feb 2021 20:13:30 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "etag",
              "value": "W/\"a6-li2T9RYtoKUz1R9KdaIOzhovpiE\""
            },
            {
              "name": "x-response-time",
              "value": "28.441ms"
            },
            {
              "name": "content-security-policy",
              "value": "child-src https://fast.wistia.net https://jumpcloud-1.wistia.com https://www.google.com https://youtube.com; connect-src 'self' https://api.segment.io https://app.launchdarkly.com https://cdn3.optimizely.com https://clientstream.launchdarkly.com https://device-cert.jumpcloud.com https://errors.client.optimizely.com https://events.launchdarkly.com https://ipv4.icanhazip.com https://logx.optimizely.com https://segment.jumpcloud.com https://solupay.transactiongateway.com https://www.google-analytics.com https://www.google.com; default-src 'self'; font-src 'self' data:; frame-ancestors https://fast.wistia.net https://jumpcloud-1.wistia.com https://www.google.com https://youtube.com; frame-src https://*.doubleclick.net https://*.duosecurity.com https://a553480116.cdn.optimizely.com/ https://fast.wistia.net https://js.driftt.com https://js.stripe.com https://jumpcloud-1.wistia.com https://solupay.transactiongateway.com https://www.google.com https://www.youtube.com https://youtube.com; img-src 'self' data: https://*.google.com https://ad.doubleclick.net https://alb.reddit.com https://assets.jumpcloud.com https://bat.bing.com https://chart.googleapis.com https://facebook.com https://googleads.g.doubleclick.net https://insight.adsrvr.org https://match.adsrvr.org https://p.typekit.net https://ping.typekit.net https://px.ads.linkedin.com https://q.stripe.com https://s3.amazonaws.com/static.jumpcloud.com/ https://stats.g.doubleclick.net https://storage.googleapis.com https://www.facebook.com https://www.google-analytics.com https://www.redditstatic.com; media-src https://fast.wistia.net https://js.driftt.com; object-src 'none'; script-src 'self' 'unsafe-eval' 'unsafe-inline' https://api.duosecurity.com https://api.stripe.com https://bat.bing.com https://cdn3.optimizely.com https://connect.facebook.com https://connect.facebook.net https://ct.capterra.com https://fast.wistia.net https://googleads.g.doubleclick.net https://js.driftt.com https://js.stripe.com https://pi.pardot.com https://px.ads.linkedin.com https://q.stripe.com https://rum.optimizely.com https://sjs.bizographics.com https://solupay.transactiongateway.com https://stats.g.doubleclick.net https://storage.googleapis.com https://tagmanager.google.com/ https://thirdparty.jumpcloud.com https://use.typekit.net https://www.google-analytics.com https://www.google.com https://www.googleadservices.com https://www.googletagmanager.com/ https://www.gstatic.com https://www.redditstatic.com; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com/ https://solupay.transactiongateway.com https://tagmanager.google.com/ https://use.typekit.net"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains;"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "x-ua-compatible",
              "value": "IE=edge"
            },
            {
              "name": "cache-control",
              "value": "no-store"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            }
          ],
          "headersSize": 3126,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-12T20:13:30.163Z",
        "time": 197,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 197
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
