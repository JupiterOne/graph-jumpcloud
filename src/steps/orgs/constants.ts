import { StepEntityMetadata } from "@jupiterone/integration-sdk-core";

export const OrgEntities: Record<
  'ORG',
  StepEntityMetadata
> = {
  ORG: {
    _type: 'jumpcloud_account',
    _class: ['Account', 'Organization'],
    resourceName: 'Organizations',
  },
};

export const PRIORITY_ORG_CACHE_KEY = 'priority_org';
