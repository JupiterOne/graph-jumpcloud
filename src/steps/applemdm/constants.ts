import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';
import { OrgEntities } from '../orgs/constants';

export const AppleEntities: Record<
  'APPLE_MDM' | 'APPLE_DEVICE',
  StepEntityMetadata
> = {
  APPLE_MDM: {
    _type: 'jumpcloud_apple_mdm',
    _class: ['Configuration'],
    resourceName: 'Configuration',
  },
  APPLE_DEVICE: {
    _type: 'jumpcloud_apple_device',
    _class: ['Device'],
    resourceName: 'Device',
  },
};
export const AppleRelationships: Record<
  'ORG_HAS_MDM_APPLE' | 'APPLE_DEVICE_HAS_MDM',
  StepRelationshipMetadata
> = {
  ORG_HAS_MDM_APPLE: {
    _type: 'jumpcloud_account_has_apple_mdm',
    _class: RelationshipClass.HAS,
    sourceType: OrgEntities.ORG._type,
    targetType: AppleEntities.APPLE_MDM._type,
  },
  APPLE_DEVICE_HAS_MDM: {
    _type: 'jumpcloud_apple_device_has_mdm',
    _class: RelationshipClass.HAS,
    sourceType: AppleEntities.APPLE_DEVICE._type,
    targetType: AppleEntities.APPLE_MDM._type,
  },
};
