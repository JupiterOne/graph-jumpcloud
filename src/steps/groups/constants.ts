import { RelationshipClass, StepEntityMetadata, StepRelationshipMetadata } from "@jupiterone/integration-sdk-core";
import { OrgEntities } from "../orgs/constants";
import { UserEntities } from "../users/constants";

export const GroupEntities: Record<
  'GROUP',
  StepEntityMetadata
> = {
  GROUP: {
    _type: 'jumpcloud_group',
    _class: ['Group'],
    resourceName: 'Group',
  },
};

export const GroupRelationships: Record<
  'ORG_HAS_GROUP' |
  'GROUP_HAS_USER',
  StepRelationshipMetadata
> = {
  ORG_HAS_GROUP: {
    _type: 'jumpcloud_account_has_group',
    _class: RelationshipClass.HAS,
    sourceType: OrgEntities.ORG._type,
    targetType: GroupEntities.GROUP._type,
  },
  GROUP_HAS_USER: {
    _type: 'jumpcloud_group_has_user',
    _class: RelationshipClass.HAS,
    sourceType: GroupEntities.GROUP._type,
    targetType: UserEntities.USER._type,
  },
};
