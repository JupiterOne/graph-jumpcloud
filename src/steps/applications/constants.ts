import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';
import { GroupEntities } from '../groups/constants';
import { UserEntities } from '../users/constants';
import { OrgEntities } from '../orgs/constants';

export const ApplicationEntities: Record<'APPLICATION', StepEntityMetadata> = {
  APPLICATION: {
    _type: 'jumpcloud_application',
    _class: ['Application'],
    resourceName: 'Application',
  },
};

export const ApplicationRelationships: Record<
  | 'USER_ASSIGNED_APPLICATION'
  | 'GROUP_ASSIGNED_APPLICATION'
  | 'ORG_HAS_APPLICATION',
  StepRelationshipMetadata
> = {
  USER_ASSIGNED_APPLICATION: {
    _type: 'jumpcloud_user_assigned_application',
    _class: RelationshipClass.ASSIGNED,
    sourceType: UserEntities.USER._type,
    targetType: ApplicationEntities.APPLICATION._type,
  },
  GROUP_ASSIGNED_APPLICATION: {
    _type: 'jumpcloud_group_assigned_application',
    _class: RelationshipClass.ASSIGNED,
    sourceType: GroupEntities.GROUP._type,
    targetType: ApplicationEntities.APPLICATION._type,
  },
  ORG_HAS_APPLICATION: {
    _type: 'jumpcloud_account_has_application',
    _class: RelationshipClass.HAS,
    sourceType: OrgEntities.ORG._type,
    targetType: ApplicationEntities.APPLICATION._type,
  },
};
