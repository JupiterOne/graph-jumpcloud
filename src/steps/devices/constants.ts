import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';
import { ApplicationEntities } from '../applications/constants';
import { UserEntities } from '../users/constants';

export const DeviceEntities: Record<'DEVICE', StepEntityMetadata> = {
  DEVICE: {
    _type: 'jumpcloud_device',
    _class: ['Device'],
    resourceName: 'Device',
  },
};

export const DeviceRelationships: Record<
  'DEVICE_INSTALLED_APPLICATION' | 'USER_HAS_DEVICE',
  StepRelationshipMetadata
> = {
  DEVICE_INSTALLED_APPLICATION: {
    _type: 'jumpcloud_device_installed_application',
    _class: RelationshipClass.INSTALLED,
    sourceType: DeviceEntities.DEVICE._type,
    targetType: ApplicationEntities.APPLICATION._type,
  },
  USER_HAS_DEVICE: {
    _type: 'jumpcloud_user_has_device',
    _class: RelationshipClass.HAS,
    sourceType: UserEntities.USER._type,
    targetType: DeviceEntities.DEVICE._type,
  },
};
