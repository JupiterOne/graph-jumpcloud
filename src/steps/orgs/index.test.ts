import { createMockStepExecutionContext } from '@jupiterone/integration-sdk-testing';
import { fetchOrgs } from '.';
import { withRecording } from '../../../test/recording';
import { integrationConfig } from '../../../test/config';
import { IntegrationConfig } from '../../config';
import { OrgEntities } from './constants';

describe('#fetchOrgs', () => {
  test('should collect data', async () => {
    await withRecording('fetchOrgs', __dirname, async () => {
      const context = createMockStepExecutionContext<IntegrationConfig>({
        instanceConfig: integrationConfig
      });

      await fetchOrgs(context);

      expect({
        numCollectedEntities: context.jobState.collectedEntities.length,
        numCollectedRelationships: context.jobState.collectedRelationships.length,
        collectedEntities: context.jobState.collectedEntities,
        collectedRelationships: context.jobState.collectedRelationships,
        encounteredTypes: context.jobState.encounteredTypes,
      }).toMatchSnapshot();

      expect(
        context.jobState.collectedEntities.filter(
          (e) => e._type === OrgEntities.ORG._type,
        ),
      ).toMatchGraphObjectSchema({
        _class: ['Account', 'Organization'],
        schema: {
          additionalProperties: false,
          properties: {
            _type: { const: 'jumpcloud_account' },
            _rawData: {
              type: 'array',
              items: { type: 'object' },
            },
            logoUrl: { type: 'string' },
          },
        },
      });
    });
  });
});
