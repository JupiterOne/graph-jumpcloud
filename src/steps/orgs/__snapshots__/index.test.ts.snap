// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchOrgs should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Account",
        "Organization",
      ],
      "_key": "60259a2ad81e1a474d4e7c83",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "_id": "60259a2ad81e1a474d4e7c83",
            "created": "2021-02-11T20:57:14.837Z",
            "displayName": null,
            "id": "60259a2ad81e1a474d4e7c83",
            "logoUrl": null,
          },
        },
      ],
      "_type": "jumpcloud_account",
      "createdOn": 1613077034837,
      "displayName": "60259a2ad81e1a474d4e7c83",
      "id": "60259a2ad81e1a474d4e7c83",
      "logoUrl": undefined,
      "name": "60259a2ad81e1a474d4e7c83",
      "webLink": "https://console.jumpcloud.com",
    },
  ],
  "collectedRelationships": [],
  "encounteredTypes": [
    "jumpcloud_account",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
